### PR TITLE
Update Thanos and Prometheus

### PR DESCRIPTION
**Pull request by pstibrany**
This PR updates Thanos and Prometheus to latest main of the respective projects.

Prometheus has renamed some packages and introduced new types for series and chunk references, which prompted lot of changes across whole codebase.

I haven't yet checked updates in dependencies.

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/api/handlers.go [hunk 1]
```diff
@@ -209,6 +209,7 @@ func NewQuerierHandler(
 		logger,
 		func(context.Context) v1.RulesRetriever { return &querier.DummyRulesRetriever{} },
 		0, 0, 0, // Remote read samples and concurrency limit.
+		false,
```
Comment (pstibrany) [inline pkg/api/handlers.go:212]: `isAgent` parameter

File: integration/alertmanager_test.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//go:build requires_docker
```
Comment (pstibrany) [inline integration/alertmanager_test.go:1]: These are added by goimport.

File: pkg/compactor/compactor_test.go [hunk 3]
```diff
@@ -421,11 +420,13 @@ func TestCompactor_ShouldIncrementCompactionErrorIfFailedToCompactASingleTenant(
 	userID := "test-user"
 	bucketClient := &bucket.ClientMock{}
 	bucketClient.MockIter("", []string{userID}, nil)
-	bucketClient.MockIter(userID+"/", []string{userID + "/01DTVP434PA9VFXSW2JKB3392D"}, nil)
+	bucketClient.MockIter(userID+"/", []string{userID + "/01DTVP434PA9VFXSW2JKB3392D", userID + "/01FN6CDF3PNEWWRY5MPGJPE3EX"}, nil)
```
Comment (pstibrany) [inline pkg/compactor/compactor_test.go:423]: Group is no longer compacted if it has only single block, so we need more blocks in compactor tests.

File: pkg/ruler/mapper_test.go [hunk 4]
```diff
@@ -395,3 +395,37 @@ func sliceContains(t *testing.T, find string, in []string) bool {
 
 	return false
 }
+
+func TestYamlFormatting(t *testing.T) {
```
Comment (pstibrany) [inline pkg/ruler/mapper_test.go:399]: Added this test to speed up debugging YAML formatting issue.

**Review by pracucci — APPROVED** [review]
Good job, LGTM!

**Pull request merged**